modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/controller/AgentDownloadController.java

```diff
@@ -78,56 +71,48 @@
-            "#server-trust-store-password=MY_PASSWORD",
-            "#cipher-suites=CIPHER1,CIPHER2,CIPHER3"
-        );
-
-        zos.write(content.getBytes());
-        zos.closeArchiveEntry();
-        zos.close();
-
-        byte[] data = baos.toByteArray();
-
-        HttpHeaders headers = new HttpHeaders();
-        headers.add(CACHE_CONTROL, "no-cache, no-store, must-revalidate");
-        headers.add(PRAGMA, "no-cache");
-        headers.add(EXPIRES, "0");
-        headers.add(CONTENT_DISPOSITION, "attachment; filename=\"" + latestAgentFileName);
-        headers.setContentLength(data.length);
-        headers.setContentType(MediaType.parseMediaType("application/zip"));
-
-        return ResponseEntity.ok()
-            .headers(headers)
-            .body(new ByteArrayResource(data));
+            try (ZipArchiveOutputStream zos = new ZipArchiveOutputStream(res.getOutputStream())) {
+                String latestAgentFileName = latestAgentPath.getFileName().toString();
+
+                res.addHeader(CACHE_CONTROL, "no-cache, no-store, must-revalidate");
+                res.addHeader(PRAGMA, "no-cache");
+                res.addHeader(EXPIRES, "0");
+                res.addHeader(CONTENT_DISPOSITION, "attachment; filename=\"" + latestAgentFileName + "\"");
+                res.setContentType("application/zip");
+
+                // Append "default.properties" to agent ZIP.
+                zos.putArchiveEntry(new ZipArchiveEntry(removeExtension(latestAgentFileName) + "/default.properties"));
+
+                String content = String.join("\n",
+                    "tokens=" + user.getToken(),
+                    "server-uri=" + currentRequestOrigin(),
+                    "#Uncomment following options if needed:",
+                    "#node-uri=http://localhost:8080",
+                    "#node-login=ignite",
+                    "#node-password=ignite",
+                    "#driver-folder=./jdbc-drivers",
+                    "#Uncomment and configure following SSL options if needed:",
+                    "#node-key-store=client.jks",
+                    "#node-key-store-password=MY_PASSWORD",
+                    "#node-trust-store=ca.jks",
+                    "#node-trust-store-password=MY_PASSWORD",
+                    "#server-key-store=client.jks",
+                    "#server-key-store-password=MY_PASSWORD",
+                    "#server-trust-store=ca.jks",
+                    "#server-trust-store-password=MY_PASSWORD",
+                    "#cipher-suites=CIPHER1,CIPHER2,CIPHER3"
+                );
+
+                zos.write(content.getBytes(UTF_8));
+                zos.closeArchiveEntry();
+
+                try (ZipFile zip = new ZipFile(latestAgentPath.toFile())) {
+                    // Make a copy of agent ZIP.
+                    zip.copyRawEntries(zos, rawEntry -> true);
+                }
+            }
+            catch (IOException ignored) {
+                // No-op.
```

Comment:
May be we should log error? I do not like "muting" errors. It will be hard to detect problems without errors in log.

Reply:
I mute only IOException (closed pipe, ...) I don't see any reason to log closed pipe or send error